core/domain/user_jobs_one_off.py

```diff
@@ -626,3 +628,63 @@ def map(item):
     def reduce(key, values):
         values.sort()
         yield (key, values)
+
+
+class VerifyProfilePictureOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that verifies various aspects of profile_picture_data_url in the
+    UserSettingsModel.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(model):  # pylint: disable=too-many-return-statements
+        if model.deleted:
+            yield ('DELETED', model.id)
+            return
+
+        if model.username is None:
+            yield ('NOT REGISTERED', model.id)
+            return
+
+        if model.profile_picture_data_url is None:
+            yield ('MISSING PROFILE PICTURE', model.id)
+            return
+
+        try:
+            profile_picture_binary = utils.convert_png_data_url_to_binary(
+                model.profile_picture_data_url)
+        except Exception:
+            yield ('WRONG PROFILE PICTURE DATA URL', model.id)
+            return
+
+        if imghdr.what(None, h=profile_picture_binary) != 'png':
+            yield ('PROFILE PICTURE NOT PNG', model.id)
+            return
+
+        try:
+            height, width = image_services.get_image_dimensions(
+                profile_picture_binary)
+        except Exception:
+            yield ('CANNOT LOAD PROFILE PICTURE', model.id)
```

Comment:
A bit of a nit, but this error seems potentially misleading. Hasn't the image already been loaded? This may indicate another issue (e.g. something specifically wrong with the file that only triggers a dimension failure).

By the way, does imghdr do any sort of format verification above? I think the try-catch here is a good idea, but I'd be slightly surprised if the two checks above this don't guarantee that the image is valid by this point.

Reply:
`get_image_dimensions` internally loads the picture via `PIL.load` so if this fails it is most probably an error with the image. Even if there is something wrong with the dimensions gathering I think it si fine to include this under same label, I will be evaluating the results of this job anyway.

> By the way, does imghdr do any sort of format verification above? I think the try-catch here is a good idea, but I'd be slightly surprised if the two checks above this don't guarantee that the image is valid by this point.

No, `imghdr` only verifies the headers of each file.

Reply:
Ah I see. Maybe add a comment here just reinforcing that, e.g.

```python
# Load the image to retrieve dimensions for later verification.
```

Or something similar.